lizmap/test/test_ogc_project_validity.py

```diff
@@ -0,0 +1,49 @@
+__copyright__ = 'Copyright 2022, 3Liz'
+__license__ = 'GPL version 3'
+__email__ = 'info@3liz.org'
+
+import unittest
+
+from qgis.core import QgsProject
+
+from lizmap.ogc_project_validity import OgcProjectValidity
+from lizmap.qgis_plugin_tools.tools.resources import plugin_test_data_path
+
+
+class TestShortNames(unittest.TestCase):
+
+    def tearDown(self) -> None:
+        QgsProject.instance().clear()
+
+    def test_shortname(self):
+        """ Test we can generate shortname. """
+        # Test shortnames
+        self.assertEqual("a_layer", OgcProjectValidity.short_name("a-layer", []))
```

Comment:
`a-layer` is valid.

Reply:
Thanks for checking.
I just added new test cases.

But with this statement, it doesn't mean `a-layer` is not valid. The current script **won't** update any existing shortnames.
So if the user has set `a-layer`, it will stay (as long as the shortname is not empty, the generation won't run)

So this is only generating a shortname from the name. And somehow, I couldn't add `-` to the `\W` regepx.
So can we keep like this ?